### PR TITLE
Fix row parallel lora layers parameters initialization bug

### PR DESCRIPTION
**Pull request by will-jl944**
### PR types
Bug fixes

### PR changes
Models

### Description

Fix the bug that in row-parallel LoRA layers, without `get_rng_state_tracker().rng_state()` specified, shards of LoRA A are initialized identically among all devices in MP group. Please refer to how the parameter is initialized in [`paddle.distributed.fleet.meta_parallel.RowParallelLinear`](https://github.com/PaddlePaddle/Paddle/blob/3017b63c1d8f84b44a538fc6056ea770243bc36c/python/paddle/distributed/fleet/layers/mpu/mp_layers.py#L657).

### CODE REVIEW

**Review by lugimzzz — DISMISSED** [review]
LGTM

File: paddlenlp/peft/lora/lora_layers.py [hunk 1]
```diff
@@ -198,14 +199,27 @@ def __init__(
...
-            shape=[self.input_size_per_partition, r],
-            dtype=self._dtype,
-            is_bias=False,
-            attr=paddle.ParamAttr(
-                initializer=nn.initializer.KaimingUniform(negative_slope=math.sqrt(5), nonlinearity="leaky_relu")
-            ),
-        )
+        if self.is_mp and paddle.in_dynamic_mode():
+            with get_rng_state_tracker().rng_state():
+                self.lora_A = self.create_parameter(
```
Comment (JunnYu) [inline paddlenlp/peft/lora/lora_layers.py]: 不要写两个if else 代码太冗余了, 直接 import contextlib  ctx = contextlib.nullcontext() if xxx else get_rng_state_tracker().rng_state()

Comment (will-jl944) [reply paddlenlp/peft/lora/lora_layers.py]: Done. Thanks.

**Review by JunnYu — DISMISSED** [review]
LGTM

**Review by gongel — DISMISSED** [review]

**Review by wawltor — APPROVED** [review]
LGTM

**Pull request merged**